### PR TITLE
TATB -35 :Show only recently completed result then create option for showing all results.

### PR DESCRIPTION
**Pull request by PrabhatJTDeveloper**
-[BUG] Getting 2 results of same quiz
      -Change the card design of all results page
      -[BUG] Data does not remain after refresh
      -Get data of completed quiz
      -implement initial letter of username as display picture
      -implement show more button in all results

### CODE REVIEW

File: src/app/component/allresults/allresults.component.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
+body {
```
Comment (charmijtdev) [inline src/app/component/allresults/allresults.component.css:1]: why to direct  body ?

Comment (charmijtdev) [reply src/app/component/allresults/allresults.component.css:1]: Please use scss format style

File: src/app/component/allresults/allresults.component.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<div class="col text-center mb-3" style="margin-top: 50px">
```
Comment (charmijtdev) [inline src/app/component/allresults/allresults.component.html:1]: Please remove inline css

File: src/app/component/allresults/allresults.component.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      return data.id == this.questionService.getUser();
+    })?.fullname;
+  }
+
+  resultData() {
+    let data: any = localStorage.getItem('result');
+    this.allResultData = JSON.parse(data).reverse();
+  }
+
+  initialLetter(userName: string) {
```
Comment (charmijtdev) [inline src/app/component/allresults/allresults.component.ts:35]: Please change this function name to something like getUserLetter

File: src/app/component/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -1,57 +1,25 @@
...
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.scss']
 })
 export class DashboardComponent implements OnInit,OnDestroy {
-  quizData = quizData;
+  quizData = {...quizData};
   title: any;
   arrayIndex=0;
```
Comment (charmijtdev) [inline src/app/component/dashboard/dashboard.component.ts:12]: Please give type number

File: src/app/component/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -1,57 +1,25 @@
...
-
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.scss']
 })
 export class DashboardComponent implements OnInit,OnDestroy {
-  quizData = quizData;
+  quizData = {...quizData};
   title: any;
```
Comment (charmijtdev) [inline src/app/component/dashboard/dashboard.component.ts:11]: please declare title type

File: src/app/component/dashboard/dashboard.component.ts [hunk 6]
```diff
@@ -1,57 +1,25 @@
...
-    // console.log("image",this.image)
-    // console.log("description",this.description)
-  }
-  // next(){
-  //   this.imagesindex++;
-  //   this.imagessindex++;
-  //   this.imagesssindex++;
-  // }
+
   quizname(title:any){
```
Comment (charmijtdev) [inline src/app/component/dashboard/dashboard.component.ts:22]: Please change name of variable

File: src/app/component/quiz/quiz.component.html [hunk 7]
```diff
@@ -14,12 +14,11 @@
       data-ride="carousel"
     >
       <form [formGroup]="myForm">
-        <button type="submit" (click)="submit()">Submit</button>
 
         <div formArrayName="form">
           <ngb-carousel
             #carousel
-            *ngIf="name"
+            *ngIf="question"
```
Comment (charmijtdev) [inline src/app/component/quiz/quiz.component.html:21]: i guess you need to check length

File: src/app/component/quiz/quiz.component.ts [hunk 8]
```diff
@@ -13,60 +13,39 @@ import { QuestionService } from '../../service/question.service';
...
-  isQuizCompleted: boolean = false;
-  findIndex: any;
-  filter: any;
   filterquestionsindex: any;
-  checkIfselctedanswer: any;
-  userName: any;
   timer: any;
   selectedQuizType: any;
   radioValue: any;
+  options: string | number | undefined;
```
Comment (charmijtdev) [inline src/app/component/quiz/quiz.component.ts:44]: options will be array not string number or undefined

File: src/app/component/quiz/quiz.component.ts [hunk 9]
```diff
@@ -199,33 +195,27 @@ export class Quizcomponent implements OnInit, OnDestroy {
           !this.FormArray.controls.at(this.questionindex)?.get('timer')
             ?.disabled
         ) {
-          this.next(this.questionindex);
+          this.nextQuestion(this.questionindex);
         }
       });
   }
 
   answer(questionindex: number, correctOptions: string) {
```
Comment (charmijtdev) [inline src/app/component/quiz/quiz.component.ts:203]: Change this function name to like submitAnswer

File: src/app/component/result/result.component.html [hunk 10]
```diff
@@ -1,88 +1,50 @@
-<div class="card shadow" id="card" style="margin-top: 80px">
-  <div class="card-body" style="padding: 0px">
-    <div class="row g-0">
-      <div class="col text-center mb-3 mt-4">
-        <span class="fs-3">Result</span>
-      </div>
-    </div>
+<div
+  class="col-lg-5 col-md-7 col-sm-12 col-xs-12"
+  style="margin: 75px auto; float: none; margin-bottom: 10px"
```
Comment (charmijtdev) [inline src/app/component/result/result.component.html:3]: still inline css

File: src/app/component/result/result.component.html [hunk 11]
```diff
@@ -1,88 +1,50 @@
...
+            <h5>Correct Answers : {{ data?.correctanswer }}</h5>
+          </span>
+          <span>
+            <h5>Wrong Answers : {{ data?.inCorrectAnswer }}</h5>
+          </span>
+        </div>
+      </div>
+      <div class="container">
+        <div class="row">
+          <div class="text-center" style="margin-bottom: 30px">
```
Comment (charmijtdev) [inline src/app/component/result/result.component.html:30]: inline css again

File: src/app/component/result/result.component.scss [hunk 12]
```diff
@@ -1,63 +1,40 @@
...
-
-
-.btn{
-    /* position: absolute; */
-    /* width: 160px;
-    left: 69px; */
-    bottom: 21px;
-}
\ No newline at end of file
+h5 {
```
Comment (charmijtdev) [inline src/app/component/result/result.component.scss:38]: wht h5 tag is directly is used

**Review by charmijtdev — CHANGES_REQUESTED** [review]

**Comment by charmijtdev** [comment]
OLD PR

File: src/app/component/allresults/allresults.component.css (hunk 1 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/allresults/allresults.component.css:1]: yes i used scss format file and removed body css

File: src/app/component/allresults/allresults.component.html (hunk 2 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/allresults/allresults.component.html:1]: Yes it is removed

File: src/app/component/allresults/allresults.component.ts (hunk 3 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/allresults/allresults.component.ts:35]: Yes i changed the function name

File: src/app/component/dashboard/dashboard.component.ts (hunk 4 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/dashboard/dashboard.component.ts:12]: it was not in use so removed it

File: src/app/component/dashboard/dashboard.component.ts (hunk 5 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/dashboard/dashboard.component.ts:11]: it was also not in use so removed it too

File: src/app/component/dashboard/dashboard.component.ts (hunk 6 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/dashboard/dashboard.component.ts:22]: yes i have changed it to startQuiz

File: src/app/component/quiz/quiz.component.html (hunk 7 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.html:21]: yes i have checked length

File: src/app/component/quiz/quiz.component.ts (hunk 9 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/quiz/quiz.component.ts:203]: Yes i changed the function name to submitAnswer

File: src/app/component/result/result.component.html (hunk 10 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/result/result.component.html:3]: i have removed the inline css

File: src/app/component/result/result.component.html (hunk 11 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/result/result.component.html:30]: removed it too

File: src/app/component/result/result.component.scss (hunk 12 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/result/result.component.scss:38]: i have removed this h5 tag styling

**Pull request closed without merging**